src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -1246,12 +1246,14 @@ protected ColumnConverter columnConverterFor(
 
       // Date is day + month + year
       case Types.DATE: {
-        return rs -> rs.getDate(col, DateTimeUtils.getTimeZoneCalendar(timeZone));
+        return rs -> rs.getDate(col,
+            DateTimeUtils.getTimeZoneCalendar(TimeZone.getTimeZone(ZoneOffset.UTC)));
```

Comment:
I think this change may be dangerous, because the behavior of this parameter is non-standard.
Also, I think that timezone conversions for dates _does_ make sense, as two different timezones experience different dates at the same instants.

Reply:
Is that true though? It is 8/27/2020 for me right now here in Seattle, but 8/28/2020 in Taiwan. Same instant in time, same millis since EPOCH, different dates.

Since the `Date` object is just a thin wrapper around millis since EPOCH, and EPOCH is defined in UTC timezone, what timezone the driver decides to use to calculate the start of day for that date is important. The reason we were seeing the hours/minutes when we get to framework Date logical type is likely because we chose the start of day to be start of the day in timezone rather than in UTC.

If we find drivers that are not implementing the timezone calculation properly, we can deal with them individually in dialects. But, I think the general case should hold that we should ask for the Date in UTC if we expect it in UTC at framework level.

Reply:
After thinking through the cases, this seems to be the safest change to implement. This does depend on the driver implementation to do the conversion, but any sufficiently strange implementations we can follow up with dialect fixes.